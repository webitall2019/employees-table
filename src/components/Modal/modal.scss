
.modal{
    display: none;
    width: 100%;
    height: 100%;
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: rgba(60, 71, 71, 0.507);
    &.visible{
        display: block;
    }
    &__btn{
        &--close{
            position: absolute;
            right: 20%;
            top: 20%;
            width: 50px;
            height: 50px;
            border-radius: 50%;
            background-color: lightcoral;
            span{
                display: block;
                width: 30px;
                height: 2px;
                background-color: #000;
                margin: 0 auto;
                transform: rotate(45deg);
                
                &:first-child{
                    position: relative;
                    top: 2px;
                    transform: rotate(-45deg);
                }
            }
            &:hover{
                transition: ease .3s;
                transform: rotateZ(90deg);
               
            }
        }
    }
}

