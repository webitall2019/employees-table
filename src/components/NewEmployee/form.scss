.form__wrap{
    padding: 120px;
    width: 80%;
    background-color: #fff;
    h1{
        text-align: center;
    }
}
.form__group{
    display: flex;
    max-width: 60%;
    margin: 0 auto;
    flex-direction: column;
    align-items: stretch;
    justify-content: space-between;
    height: 100%;
    input, select{
       
        padding: 10px;
        margin-bottom: 20px;
    }
    @media screen and (max-width: 500px) {
        max-width: initial;
    }
}