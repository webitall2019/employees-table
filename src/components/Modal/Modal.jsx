import React from "react";
import "./modal.scss";

const Modal = ({ children, modalHandler }) => (
  <div className="modal" role="showModal" data-testid="modalTestId">
    <button onClick={modalHandler} className="modal__btn--close">
      <span></span>
      <span></span>
    </button>
    {children}
  </div>
);

export default Modal;
