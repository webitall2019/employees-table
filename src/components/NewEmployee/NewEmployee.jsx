import { useDispatch } from "react-redux";
import React, { useState, useEffect } from "react";
import "./form.scss";

const NewEmployee = () => {
  const [newEmployee, setNewEmployee] = useState({});
  const [name, setName] = useState("");
  const [tenure, setAge] = useState("");
  const [jobTitle, setJob] = useState("CEO");
  const [gender, setGender] = useState("Male");
  const dispatch = useDispatch();

  useEffect(() => {
    setNewEmployee({
      name,
      tenure,
      jobTitle,
      gender,
    });
  }, [name, tenure, jobTitle, gender]);

  const submithandler = async (e) => {
    e.preventDefault();
    dispatch({ type: "newUser", payload: newEmployee });
  };

  return (
    <div className="form__wrap">
      <h1>Become one of the team</h1>
      <form onSubmit={submithandler}>
        <div className="form__group">
          <input
            data-testid="userName"
            required
            placeholder="your name"
            id="name"
            type="text"
            name="name"
            value={name}
            onChange={(e) => setName(e.target.value)}
          />
          <input
            data-testid="userAge"
            required
            placeholder="your age"
            type="number"
            name="tenure"
            value={tenure}
            onChange={(e) => setAge(e.target.value)}
          />
          <select
            data-testid="jobTitle"
            name="jobTitle"
            onChange={(e) => setJob(e.target.value)}
            required
          >
            <option value="CEO">CEO</option>
            <option value="Developer">Developer</option>
            <option value="Marketing Specialist">Marketing Specialist</option>
            <option value="CFO">CFO</option>
            <option value="QA">QA</option>
          </select>
          <select
            data-testid="gender"
            name="gender"
            onChange={(e) => setGender(e.target.value)}
            required
          >
            <option value="Male">Male</option>
            <option value="Female">Female</option>
          </select>
          <button type="submit">submit</button>
        </div>
      </form>
    </div>
  );
};
export default NewEmployee;
