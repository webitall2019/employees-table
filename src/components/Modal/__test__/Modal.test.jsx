import React from "react";
import { act, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import store from "../../../store/store";
import Modal from "../Modal";
import NewEmployee from "../../NewEmployee/NewEmployee";

describe("<Modal/> component ", () => {
  it("check to be in document", () => {
    const { getByTestId } = render(<Modal />);
    const modal = screen.getByTestId("modalTestId");
    expect(modal).toBeInTheDocument();
  });
  it("should match snapshot WHEN rendered", async () => {
    const modalHandler = jest.fn();
    const { asFragment } = render(
      <Modal modalHandler={modalHandler}>
        <Provider store={store}>
          <NewEmployee />
        </Provider>
      </Modal>
    );
    act(async () => await expect(asFragment()).toMatchSnapshot());
  });
});
