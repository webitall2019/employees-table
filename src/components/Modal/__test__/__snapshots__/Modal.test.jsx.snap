// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> component  should match snapshot WHEN rendered 1`] = `
<DocumentFragment>
  <div
    class="modal"
    data-testid="modalTestId"
    role="showModal"
  >
    <button
      class="modal__btn--close"
    >
      <span />
      <span />
    </button>
    <div
      class="form__wrap"
    >
      <h1>
        Become one of the team
      </h1>
      <form>
        <div
          class="form__group"
        >
          <input
            data-testid="userName"
            id="name"
            name="name"
            placeholder="your name"
            required=""
            type="text"
            value=""
          />
          <input
            data-testid="userAge"
            name="tenure"
            placeholder="your age"
            required=""
            type="number"
            value=""
          />
          <select
            data-testid="jobTitle"
            name="jobTitle"
            required=""
          >
            <option
              value="CEO"
            >
              CEO
            </option>
            <option
              value="Developer"
            >
              Developer
            </option>
            <option
              value="Marketing Specialist"
            >
              Marketing Specialist
            </option>
            <option
              value="CFO"
            >
              CFO
            </option>
            <option
              value="QA"
            >
              QA
            </option>
          </select>
          <select
            data-testid="gender"
            name="gender"
            required=""
          >
            <option
              value="Male"
            >
              Male
            </option>
            <option
              value="Female"
            >
              Female
            </option>
          </select>
          <button
            type="submit"
          >
            submit
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
